### PR TITLE
Add com.riverbankcomputing.PyQt.BaseApp

### PR DESCRIPTION
**Pull request by JakobDev**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

This is a BaseApp which contains PyQt. It used by many Python Programs. It is not easy to build and take some time to build (around 10 Minutes).  This BaseApp currently has PyQt6. PyQt5 may be added in the future. I'm also planing to make a second BaseApp which is based on [io.qt.qtwebengine.BaseApp](https://github.com/flathub/io.qt.qtwebengine.BaseApp) and contains Qt Web Engine.

### CODE REVIEW

**Comment by JakobDev** [comment]
bot, build com.riverbankcomputing.PyQt.BaseApp

**Comment by hfiguiere** [comment]
how do you provide an that needs both QtWebEngine and PyQt?

File: com.riverbankcomputing.PyQt.BaseApp.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+app-id: com.riverbankcomputing.PyQt.BaseApp
+runtime: org.kde.Platform
+runtime-version: "6.2"
+sdk: org.kde.Sdk
+command: /usr/bin/python
```
Comment (hfiguiere) [inline com.riverbankcomputing.PyQt.BaseApp.yaml:5]: Command is NEVER an aboslute path

Comment (JakobDev) [reply com.riverbankcomputing.PyQt.BaseApp.yaml:5]: This is a BaseApp and never meant to run directly. This is just a placeholder.

**Comment by hfiguiere** [comment]
so you want to maintain 2^4 packages?

**Comment by JakobDev** [comment]
> so you want to maintain 2^4 packages?

All of my programs use PyQt, one PyQt with QtWebengine. This means I have to maintain this anyway. It might be better, to use a separate yml with contains PyQt, so I have to edit only this file and could use git submodules. For PyQt5: Since Qt5 itself is EOL, if you are not a paying customer which has longer support, there will not be much to do here. Only Bugfix updates, which can be delegated to flatpak-external-data-checker.

**Comment by hfiguiere** [comment]
see https://github.com/flathub/shared-modules/pull/145

Also if these modules were not so impossible to build reliably, and if the tarball didn't disappear. but that's an upstream problem.

File: com.riverbankcomputing.PyQt.BaseApp.yaml (hunk 1 above)
Comment (barthalion) [reply com.riverbankcomputing.PyQt.BaseApp.yaml:5]: I think `command` field is optional though. You can just remove it.

**Comment by barthalion** [comment]
Maintaining two base apps doesn't seem reasonable. I would prefer a single BaseApp with both PyQt and QtWebEngine.

Also please sync with @tinywrkb about your work.

**Comment by tinywrkb** [comment]
@JakobDev my WIP PyQtWebEngine baseapp is [here](https://github.com/tinywrkb/flathub/tree/com.riverbankcomputing.PyQtWebEngine.BaseApp).  
I'm in the middle of polishing it, and sending a new app PR is next on my todo list, but not all the bits that I want in are in place yet.

**Comment by JakobDev** [comment]
I'm closing this in favour of #2998

**Pull request closed without merging**